int32   direction
float64 speed
